org.eclipse.scout.rt.ui.html/src/main/js/scout/util/DragAndDropHandler.js

```diff
@@ -0,0 +1,126 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     BSI Business Systems Integration AG - initial API and implementation
+ ******************************************************************************/
+scout.DragAndDropHandler = function(options) {
+  options = options || {};
+  this.additionalDropProperties = null;
+  this.allowedTypes = null;
+  this.dropMaximumSize = null;
+  this.target = null;
+  this.onDrop = null;
+
+  $.extend(this, options);
+  this.supportedScoutTypes = scout.arrays.ensure(options.supportedScoutTypes);
+  this._onDragEnterHandler = this._onDragEnter.bind(this);
+  this._onDragOverHandler = this._onDragOver.bind(this);
+  this._onDropHandler = this._onDrop.bind(this);
+};
+
+scout.DragAndDropHandler.prototype.install = function($element, selector) {
+  if (this.$element) {
+    throw new Error('Already installed.');
+  }
+  this.selector = selector;
+  this.$element = $element;
+  this.$element.on('dragenter', this.selector, this._onDragEnterHandler)
+    .on('dragover', this.selector, this._onDragOverHandler)
+    .on('drop', this.selector, this._onDropHandler);
+};
+
+scout.DragAndDropHandler.prototype.uninstall = function() {
+  this.$element.off('dragenter', this.selector, this._onDragEnterHandler)
+    .off('dragover', this.selector, this._onDragOverHandler)
+    .off('drop', this.selector, this._onDropHandler);
+  this.$element = null;
+  this.selector = null;
+};
+
+scout.DragAndDropHandler.prototype._onDragEnter = function(event) {
+  this._onDragEnterOrOver(event);
+};
+
+scout.DragAndDropHandler.prototype._onDragOver = function(event) {
+  this._onDragEnterOrOver(event);
+};
+
+scout.DragAndDropHandler.prototype._onDragEnterOrOver = function(event) {
+  // set dropEffect to copy. otherwise outlook will move dropped mails into the deleted files folder.
+  // see: https://bugs.chromium.org/p/chromium/issues/detail?id=322605#c33
+  event.originalEvent.dataTransfer.dropEffect = 'copy';
+  scout.dragAndDrop.verifyDataTransferTypesScoutTypes(event, this.supportedScoutTypes, this.dropType());
+};
+
+scout.DragAndDropHandler.prototype._onDrop = function(event) {
+  if (this.supportedScoutTypes.indexOf(scout.dragAndDrop.SCOUT_TYPES.FILE_TRANSFER) >= 0 &&
+    (this.dropType() & scout.dragAndDrop.SCOUT_TYPES.FILE_TRANSFER) === scout.dragAndDrop.SCOUT_TYPES.FILE_TRANSFER && // NOSONAR
+    scout.dragAndDrop.dataTransferTypesContainsScoutTypes(event.originalEvent.dataTransfer, scout.dragAndDrop.SCOUT_TYPES.FILE_TRANSFER)) {
+    if (!this.onDrop ||
+      !(event.originalEvent.dataTransfer.files instanceof FileList)) {
+      return;
+    }
+    var files = Array.from(event.originalEvent.dataTransfer.files);
+    if (scout.arrays.empty(files)) {
+      return;
+    }
+    try {
+      this.validateFiles(files);
+    } catch (error) {
+      this._validationFailed(files, error);
+      return;
+    }
+    event.stopPropagation();
+    event.preventDefault();
+    var formattedDropEvent = {
```

Comment:
Why is this event "formatted"? I'd simply inline this object into the onDrop call